numba/pycc/platform.py

```diff
@@ -26,40 +28,28 @@ def get_configs(arg):
 find_shared_ending = functools.partial(get_configs, 0)
 find_pyext_ending = functools.partial(get_configs, 1)
 
-@contextmanager
-def _gentmpfile(suffix):
-    # windows locks the tempfile so use a tempdir + file, see
-    # https://github.com/numba/numba/issues/3304
-    try:
-        tmpdir = mkdtemp()
-        ntf = open(os.path.join(tmpdir, "temp%s" % suffix), 'wt')
-        yield ntf
-    finally:
-        try:
-            ntf.close()
-            os.remove(ntf)
-        except:
-            pass
-        else:
-            os.rmdir(tmpdir)
 
 def _check_external_compiler():
     # see if the external compiler bound in numpy.distutil is present
     # and working
     compiler = new_compiler()
     customize_compiler(compiler)
     for suffix in ['.c', '.cxx']:
+        tmpdir = mkdtemp()
+        # windows locks the tempfile so use a tempdir + file, see
+        # https://github.com/numba/numba/issues/3304
+        with open(os.path.join(tmpdir, "temp%s" % suffix), 'wt') as ntf:
+            simple_c = "int main(void) { return 0; }"
+            ntf.write(simple_c)
         try:
-            with _gentmpfile(suffix) as ntf:
-                simple_c = "int main(void) { return 0; }"
-                ntf.write(simple_c)
-                ntf.flush()
-                ntf.close()
-                # *output_dir* is set to avoid the compiler putting temp files
-                # in the current directory.
-                compiler.compile([ntf.name], output_dir=gettempdir())
-        except Exception: # likely CompileError or file system issue
+            # *output_dir* is set to avoid the compiler putting temp files
+            # in the current directory.
+            compiler.compile([ntf.name], output_dir=Path(ntf.name).anchor)
```

Comment:
This is the only change in behaviour. Could you please revert all the other changes except this part?

Reply:
ok should be done now.